numpy/f2py/crackfortran.py

```diff
@@ -832,7 +832,7 @@ def appenddecl(decl, decl2, force=1):
         decl = {}
     if not decl2:
         return decl
-    if decl is decl2:
+    if decl == decl2:
```

Comment:
I'd be inclined to change this line in another PR, and only if a failing test can be constructed with it in its current state.

The rest of this PR is uncontroversial, and would be good to get in